foqus_lib/foqus.py

```diff
@@ -0,0 +1,760 @@
+        showOpt: Show the optimzation tab
+        showBasicDataTab: Show the Basic Data tab
+        ts: A testing script to automatiacally run when GUI starts.
+    """
+    import foqus_lib.gui.main.mainWindow as MW
+    if app == None:
+        app = QApplication(sys.argv)
+    #create main window and start application loop
+    makeSplash()
+    if showSplash:
+        #add timer to show splash
+        splashScr[0] = QtCore.QTimer()
+        splashScr[0].timeout.connect(hideSplash)
+        # splash_timeout_ms is how long to show splash in ms
+        # it is set in the first code line of this file
+        splashScr[0].start(splash_timeout_ms)
+        splashScr[1].setWindowFlags(
+            splashScr[1].windowFlags()|
+            QtCore.Qt.WindowStaysOnTopHint)
+        splashScr[1].show()
+
+    mainWin = MW.mainWindow(
+        "FOQUS", # window title
+        800, # width
+        600, # height
+        dat, # FOQUS session data
+        splashScr[1], #splash screen to use for about
+        showUQ=showUQ,
+        showOpt=showOpt,
+        showBasicData=showBasicData,
+        showSDOE=showSDOE,
+        ts=ts)
+    mainWin.app = app
+    app.exec_()
+    return mainWin
+
+def logException(etype, evalue, etrace):
+    """
+    A function to assign to sys.excepthook to cause unhandled exceptions to go
+    to the log file instead of stderr.  If GUI is started this will also attempt
+    to show unhandled exceptions in a dialod box, so user is aware.
+    """
+    try:
+        hideSplash()
+    except:
+        pass
+    try:
+        logging.getLogger("foqus." + __name__).critical(
+            "unhandled exception",
+            exc_info=(etype, evalue, etrace))
+        if foqus_application:
+            # just incase the exception happend when
+            # cursor was set to waiting type set to normal
+            foqus_application.restoreOverrideCursor()
+            # Now show error in a message box
+            msgBox = QMessageBox()
+            msgBox.setWindowTitle("Error")
+            msgBox.setText("Unhandled Exception:")
+            msgBox.setInformativeText(
+                ''.join(
+                    traceback.format_exception(etype, evalue, etrace)))
+            msgBox.exec_()
+    except Exception as e:
+        logging.getLogger("foqus." + __name__).exception(
+            "unhandled exception: If you see this there was some"\
+            " problem with unhandled exception logging. ")
+
+def signal_handler(signal, frame):
+    """
+    A signal handler to cause a siginal to raise a keyboardinterupt exception.
+    Used to override a default signal like SIGINT so the FOQUS consumer process
+    can shutdown cleanly. The FOQUS consumer catches the keyboardinterupt
+    exception as one (slighlty unreliable) way to shut down.  Seems ctrl-c
+    causes keyboard interupt exception and SIGINT signal, hense need to change
+    SIGINT handler.
+    """
+    raise KeyboardInterrupt()
+
+def main():
```

Comment:
Add ```global dat``` here, since you moved this into a main function.  Then this should work.  I guess we really need to rework foqus.py, but this is the easy fix for now.